### ISSUE TITLE
*** not working as a snippet trigger

### ISSUE THREAD

**Issue by adamreisnz** [microsoft/vscode#75254]
Issue Type: <b>Bug</b>

Define snippet:

```
{
  "Divider": {
		"scope": "javascript,typescript",
		"prefix": "***",
		"body": [
			"/**************************************************************************",
			" * ${1:Label}",
			" ***/",
		],
		"description": "Divider"
	},
}
```

Try to toggle it, snippet won't work.

VS Code version: Code - Insiders 1.36.0-insider (7a5bca9fafbbb23de289179be95e9c4832540c02, 2019-06-06T11:36:26.351Z)
OS version: Darwin x64 18.6.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-3770 CPU @ 3.40GHz (8 x 3400)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>oop_rasterization: unavailable_off<br>protected_video_decode: unavailable_off<br>rasterization: unavailable_off<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|8.00GB (0.02GB free)|
|Process Argv|--verbose|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (8)</summary>

Extension|Author (truncated)|Version
---|---|---
project-manager|ale|10.5.2
vscode-eslint|dba|1.9.0
EditorConfig|Edi|0.13.0
file-icons|fil|1.0.17
multi-command|ryu|1.3.0
vscode-fileutils|sle|2.14.5
reopenclosedtab|uyi|1.1.0
vscode-todo-highlight|way|1.0.4

(1 theme extensions excluded)

</details>

**Comment by jrieken** [comment]
> Try to toggle it, snippet won't work.

Not sure what "toggle" is supposed to mean but this snippet works for me

<img width="665" alt="Screenshot 2019-06-12 at 11 35 33" src="https://user-images.githubusercontent.com/1794099/59340387-34ffa480-8d06-11e9-8c8e-ba3458055fbc.png">

**Comment by adamreisnz** [comment]
![snippet](https://user-images.githubusercontent.com/490562/59343448-057d8600-8d60-11e9-8744-598812b75835.gif)

By "not toggling" I mean that it's not showing a context menu to allow me to toggle the snippet with <kbd>TAB</kbd>.

Compare to when I type `con`, it's showing the snippet and the context menu with other options, but after I type `***` it doesn't show anything, and pressing <kbd>TAB</kbd> just tabs away. So no way to trigger the snippet.

**Comment by jrieken** [comment]
So, ⬇️ isn't how it works for you? For me the `***` works like any other snippets, e.g `foreach`. Any special settings you have configured? 

![Jun-12-2019 12-29-15](https://user-images.githubusercontent.com/1794099/59344145-c45c8600-8d0d-11e9-98c6-d0da8d49aed1.gif)

**Comment by adamreisnz** [comment]
That's right yes, not showing any snippet suggestions for the `***` combination.

I only started using VSCode a few days ago, but can't see anything obvious in the configuration that might prevent this particular snippet from triggering.

All the snippets I use have the same scope, so no discrepancy there either.

Here is my full config (minus theme/syntax coloring), do you think one of  them might be affecting this? Happy to run some tests if you can point out a few settings that you think may be causing this.

```
    "search.location": "panel",
    "window.zoomLevel": 0,
    "workbench.colorTheme": "Monokai Dimmed",
    "workbench.iconTheme": "file-icons-colourless",
    "editor.fontSize": 16,
    "search.quickOpen.includeHistory": false,
    "workbench.statusBar.feedback.visible": false,
    "window.openFoldersInNewWindow": "on",
    "eslint.autoFixOnSave": true,
    "eslint.run": "onSave",
    "editor.wordWrap": "on",
    "editor.rulers": [80],
    "editor.renderIndentGuides": true,
    "editor.formatOnPaste": true,
    "workbench.list.automaticKeyboardNavigation": false,
    "workbench.editor.enablePreviewFromQuickOpen": false,
    "workbench.editor.tabSizing": "shrink",
    "workbench.startupEditor": "newUntitledFile",
    "javascript.suggestionActions.enabled": false,
    "editor.occurrencesHighlight": false,
    "explorer.confirmDelete": false,
    "editor.minimap.maxColumn": 80,
    "explorer.autoReveal": false,
    "explorer.decorations.badges": false,
    "explorer.openEditors.visible": 0,
    "editor.multiCursorModifier": "ctrlCmd",
    "editor.snippetSuggestions": "top",
    "javascript.preferences.quoteStyle": "single",
    "javascript.suggest.completeJSDocs": false,
    "typescript.suggest.completeJSDocs": false,
    "window.newWindowDimensions": "maximized",
    "html.format.enable": false,
    "html.autoClosingTags": false,
    "editor.quickSuggestionsDelay": 0,
    "typescript.preferences.quoteStyle": "single",
    "explorer.confirmDragAndDrop": false,
    "editor.defaultFormatter": "dbaeumer.vscode-eslint"
```

**Comment by jrieken** [comment]
Any additional workspace configurations? In what file type are you trying this?

**Comment by adamreisnz** [comment]
Just a plain Javascript file, no other configurations.

**Comment by jrieken** [comment]
Ok, I saw this but I can still not reliably reproduce this. It seems that the suggestion request just hangs at times...

**Comment by jrieken** [comment]
I have a theory. @adamreisnz Can you confirm that stuff recovers when pressing Ctrl+Space once, e.g be in a broken state, press Ctrl+Space (see/filter to the ***-suggestion), delete the whatever has been typed, type * -> nothing

**Comment by adamreisnz** [comment]
When I activate <kbd>CTRL</kbd-<kbd>SPACE</kbd>, the dropdown does show up, but it doesn't contain the `***` snippet, just my other one for the `/*` trigger:

![image](https://user-images.githubusercontent.com/490562/59466874-7026e280-8e82-11e9-819b-cb744e0ab9d4.png)

As an experiment, I removed that snippet to see if it was affecting it, but it doesn't seem to be the case. Even without it, it still doesn't show the `***` snippet/suggestion:

![image](https://user-images.githubusercontent.com/490562/59466944-9cdafa00-8e82-11e9-908a-bcf36a3eb54a.png)

As another experiment, I tried with the prefix `---` instead, which yielded the same result.

Changed to the prefix `divider`, and that does work:

![image](https://user-images.githubusercontent.com/490562/59467033-d3b11000-8e82-11e9-8fcd-5ccc55d2bcb9.png)

Tried a few more variations:

![image](https://user-images.githubusercontent.com/490562/59467077-ed525780-8e82-11e9-9928-d44ca2719ef8.png)

![image](https://user-images.githubusercontent.com/490562/59467103-f80cec80-8e82-11e9-8005-705839005178.png)

![image](https://user-images.githubusercontent.com/490562/59467153-107d0700-8e83-11e9-8bca-cc8427a2cae1.png)

However, the prefix `***d` with three asterisks didn't work.

So seems like something about the `*` and `-` characters if using 3 or more it seems to stop working.

**Comment by jrieken** [comment]
> So seems like something about the *

Yeah, those are characters that trigger completions themselves. Turned out this was the same issue as #75402. Both are now fixed

**Comment by adamreisnz** [comment]
Excellent, thank you
